documentation/jetty/modules/code/examples/src/main/java/org/eclipse/jetty/docs/programming/ArchitectureDocs.java

```diff
@@ -75,4 +79,78 @@ public void virtualVirtualThreads()
         HttpClient httpClient = new HttpClient(new HttpClientTransportOverHTTP(clientConnector));
         // end::virtualVirtual[]
     }
+
+    public static class EitherTask implements Invocable.Task
+    {
+        private final TryExecutor executor;
+        private final Deque<Invocable.Task> subTasks = new ConcurrentLinkedDeque<>();
+
+        public EitherTask(TryExecutor executor)
+        {
+            this.executor = executor;
+        }
+
+        @Override
+        public InvocationType getInvocationType()
+        {
+            return InvocationType.EITHER;
+        }
+
+        public void offer(Task task)
+        {
+            subTasks.add(task);
+        }
+
+        @Override
+        public void run()
+        {
+            if (Invocable.isNonBlockingInvocation())
+            {
+                for (Task subTask = subTasks.pollFirst(); subTask != null; subTask = subTasks.pollFirst())
+                {
+                    switch (Invocable.getInvocationType(subTask))
+                    {
+                        case NON_BLOCKING, EITHER ->
+                            // Do not defer a NON-BLOCKING or EITHER task
+                            subTask.run();
+                        case BLOCKING ->
+                            // Defer the BLOCKING task as this thread may not block
+                            executor.execute(subTask);
+                    }
+                }
+            }
+            else
+            {
+                for (Task subTask = subTasks.pollFirst(); subTask != null; subTask = subTasks.pollFirst())
+                {
+                    switch (Invocable.getInvocationType(subTask))
+                    {
+                        case NON_BLOCKING ->
+                            // Do not defer a NON-BLOCKING task
+                            subTask.run();
+                        case EITHER ->
+                        {
+                            if (executor.tryExecute(this))
```

Comment:
`executor.tryExecute()` is merely an optimization, as if the code only was what's in the `else` branch it would still be correct, right?

I'm wondering if it's wise to include that optimization as part of the documentation, and if so, if that shouldn't be specified?

Reply:
We need to tryExecute in both branches to show that we can block the calling thread, but only if we do not head of line block the processing of other sub tasks.

Reply:
I've removed tryExecute from the example